LogicalTypes.md

```diff
@@ -366,3 +366,27 @@ optional group my_map (MAP_KEY_VALUE) {
 }
 ```
 
+### Unions
+
+A Union type annotates data stored as a Group.
+It describes the different possible types under the same field name.
+The names of the fields in the annotated Group are not important in such a case, but
+as a convention the type names are used.
+All fields of the Group must be optional and exactly one is defined for each instance of the group.
+If more than one is defined the behavior is undefined and may changed depending on the projection applied.
+A Union can not contain null but can be null itself if in an optional field.
+
+// Union<String, Integer, Boolean> (nullable union of either String, Integer or Boolean)
```

Comment:
I think it would be more clear to use "Union of null, String, Integer, or Boolean" because the intent is not to have a Union container that may itself be null, although that is how it is stored. If we're going with Java-ish types for clarity, then maybe it's more accurate to use "Void | String | Integer | Boolean".